### PR TITLE
GEODE-9255: Remove remaining uses of ByteArrayWrapper

### PR DESCRIPTION
**Pull request by DonalEvans**
- Make RedisKey implement the necessary methods that ByteArrayWrapper
 was previously implementing
 - Remove all uses of ByteArrayWrapper

Authored-by: Donal Evans <doevans@vmware.com>

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [N/A] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisKey.java [hunk 1]
```diff
@@ -63,6 +66,78 @@ public int getBucketId() {
...
+   * This equals is not symmetric and therefore not transitive, because a String with the same
+   * underlying bytes is considered equal. Clearly calling {@link String#equals(Object)} would not
+   * yield the same result
+   */
+  @Override
+  public boolean equals(Object other) {
+    if (other instanceof RedisKey) {
+      return Arrays.equals(value, ((RedisKey) other).value);
+    } else if (other instanceof String) {
+      return Arrays.equals(value, Coder.stringToBytes((String) other));
```
Comment (jdeppe-pivotal) [inline geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisKey.java:136]: I'm not 100%, but I think this check can be removed (at least all integration tests pass without it). I can't think where we might be comparing a `RedisKey` to a string.

**Review by jdeppe-pivotal — CHANGES_REQUESTED** [review]

File: geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisKey.java (hunk 1 above)
Comment (DonalEvans) [reply geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisKey.java:136]: I think this is leftover code from when ByteArrayWrapper was being used, and that we being compared to strings. I'll remove it.

**Review by jdeppe-pivotal — APPROVED** [review]

File: geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisKey.java [hunk 2]
```diff
@@ -21,23 +21,26 @@
...
 import org.apache.geode.internal.serialization.DataSerializableFixedID;
 import org.apache.geode.internal.serialization.DeserializationContext;
 import org.apache.geode.internal.serialization.KnownVersion;
 import org.apache.geode.internal.serialization.SerializationContext;
 import org.apache.geode.redis.internal.executor.cluster.CRC16;
 import org.apache.geode.redis.internal.executor.cluster.RedisPartitionResolver;
+import org.apache.geode.redis.internal.netty.Coder;
 
-public class RedisKey extends ByteArrayWrapper implements DataSerializableFixedID {
+public class RedisKey implements DataSerializableFixedID, Comparable<RedisKey> {
```
Comment (sabbey37) [inline geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisKey.java:35]: Have you found that we still need to implement `Comparable`?

Comment (DonalEvans) [reply geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisKey.java:35]: I think it's likely that we don't. I'll remove this and see if anything breaks in pre-checkin.

File: geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/GeodeRedisService.java [hunk 3]
```diff
@@ -61,9 +60,6 @@ private void registerDataSerializables() {
     InternalDataSerializer.getDSFIDSerializer().registerDSFID(
         DataSerializableFixedID.REDIS_KEY,
         RedisKey.class);
-    InternalDataSerializer.getDSFIDSerializer().registerDSFID(
-        DataSerializableFixedID.REDIS_BYTE_ARRAY_WRAPPER,
```
Comment (upthewaterspout) [inline geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/GeodeRedisService.java:65]: I think we should remove this constant from `DataSerializableFixedID` as well?

**Review by upthewaterspout — APPROVED** [review]

File: geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/GeodeRedisService.java (hunk 3 above)
Comment (DonalEvans) [reply geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/GeodeRedisService.java:65]: Good catch, thanks!

**Review by Bill — APPROVED** [review]
approved!

**Pull request merged**